File: integration-tests/src/test/java/dev/capylang/test/LocalVisibilityTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;
import dev.capylang.test.localVisibility.PackageLocalSupport;
import dev.capylang.test.localVisibility.PackagePeer;
import dev.capylang.test.localVisibility.child.PackageChild;

import static org.assertj.core.api.Assertions.assertThat;

class LocalVisibilityTest {
    @Test
    void samePackageModuleCanUseItsOwnLocalMembers() {
        assertThat(PackageLocalSupport.ownPackageValue(1)).isEqualTo(6);
        assertThat(PackageLocalSupport.ownPackageValue(3)).isEqualTo(8);
    }

    @Test
    void peerModuleCanUseLocalMembersFromSamePackage() {
        assertThat(PackagePeer.peerCanUseLocalMembers(1)).isEqualTo(6);
        assertThat(PackagePeer.peerCanUseLocalMembers(3)).isEqualTo(8);
    }

    @Test
    void childPackageCanUseLocalMembersFromParentPackage() {
        assertThat(PackageChild.childCanUseLocalMembers(1)).isEqualTo(6);
        assertThat(PackageChild.childCanUseLocalMembers(3)).isEqualTo(8);
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/LocalVisibility.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('LocalVisibility', () => {
    const packageLocal = generatedModule('dev/capylang/test/localVisibility/PackageLocalSupport.js');
    const packagePeer = generatedModule('dev/capylang/test/localVisibility/PackagePeer.js');
    const packageChild = generatedModule('dev/capylang/test/localVisibility/child/PackageChild.js');
    assert.equal(packageLocal.ownPackageValue(1), 6);
    assert.equal(packagePeer.peerCanUseLocalMembers(3), 8);
    assert.equal(packageChild.childCanUseLocalMembers(1), 6);
});
